tests/console/nvme_checks.pm

```diff
@@ -0,0 +1,116 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Smoke tests for NVM Express Controller.
+#
+# - Install nvme-cli which is a tool to manage NVM Express
+# - Check sys path that should be hold details
+# - Check if we have any NVMe controllers on the PCI bus
+# - Check if we have NVMe devices
+# - Check if we have namespace
+# - Send Identify Admin Command to the NVMe controller
+# - Issue a Read command both using the helper and manually
+# - Compare it to what we have with a regular dd
+#
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base "installbasetest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use scheduler 'get_test_suite_data';
+
+sub _smoke_checks {
+    my $nvm_test_data = shift;
+    assert_script_run("lspci -nn | grep -i nvm");
+    assert_script_run("test -d /sys/class/nvme");
+    assert_script_run("test -c /dev/nvme || echo 'nvm character devices not found'");
+
+    my @nvm_partitions = split(/\n/, script_output("ls -l /dev/$nvm_test_data->{nvm_disk}*"));
+    foreach (@nvm_partitions) {
+        assert_script_run("test -b $_ || echo 'nvm is not a block device'");
+    }
+}
+
+sub _exercise_nvme_commands_and_validate_output {
+    my $nvm_test_data = shift;
+    # send Identify Admin Command to the NVMe controller.
+    assert_script_run("nvme id-ctrl -H /dev/$nvm_test_data->{nvm_char_device}");
+    save_screenshot;
+    my $MODE       = '\/dev\/' . $nvm_test_data->{nvm_char_device};
+    my $SN         = $nvm_test_data->{sn};
+    my $MODEL      = $nvm_test_data->{model};
+    my $NAMESPACE  = $nvm_test_data->{namespace_count};
+    my $DISK_SIZE  = $nvm_test_data->{nvm_disk_size};
+    my $BLOCK_SIZE = $nvm_test_data->{nvm_block_size};
+    my $list_regex = qr/${MODE}\s+$SN\s+${MODEL}\s+${NAMESPACE}\s+${DISK_SIZE}\s+GB.*\d+\s+GB\s+${BLOCK_SIZE}/;
```

Comment:
Instead of declaring so many variables which is a bad thing for maintenance, specially when they don't have any kind of grouping and someone add more code in the middle making hard to keep track of them. Have you check [this other way](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/repo_tools.pm#L468), it is quite neat, I would recommend it.

Reply:
i believe that the two approaches are interchangeable. Very nice also the suggestion. the subroutine is relative small so i dont think it will cause any confusion. The only reason for the introduction of the variables here is to make it clearer an avoid a long line in the regex as much as possible.